src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,181 @@
+    shooterLeftMotor.getConfigurator().apply(shooterMotorConfig);
+    shooterMotorConfig.MotorOutput.Inverted = InvertedValue.CounterClockwise_Positive;
+    shooterRightMotor.getConfigurator().apply(shooterMotorConfig);
+
+    // Get the through-bore-encoder absolute encoder
+    actuatorEncoder = actuatorMotor.getAbsoluteEncoder(kDutyCycle);
+    actuatorEncoder.setInverted(true);
+    actuatorEncoder.setAverageDepth(64);
+    pidController = actuatorMotor.getPIDController();
+    pidController.setFeedbackDevice(actuatorEncoder);
+
+  }
+
+  private static SparkPIDController indexerMotorConfig(CANSparkMax sparkMax, boolean invert) {
+    SparkPIDController pidController = sparkMax.getPIDController();
+    sparkMax.restoreFactoryDefaults();
+    sparkMax.enableVoltageCompensation(12);
+    sparkMax.setOpenLoopRampRate(0.1);
+    sparkMax.setClosedLoopRampRate(0.1);
+    sparkMax.setInverted(invert);
+    pidController.setP(IndexerConstants.BELT_kP);
+    pidController.setFF(0.00009);
+    sparkMax.getEncoder();
+    sparkMax.burnFlash();
+    sparkMax.setIdleMode(IdleMode.kCoast);
+    return pidController;
+
+  }
+
+  public void spinShooterWheel(double shooterRPS) {
+    shooterLeftMotor.setControl(shooterMotorVelocity.withVelocity(shooterRPS));
+    shooterRightMotor.setControl(shooterMotorVelocity.withVelocity(shooterRPS));
+  }
+
+  public Command sysIdShooterMotorQuasiCommand(Direction direction) {
+    return shooterMotorSysIdRoutine.quasistatic(direction).withName("SysId Shooter Motors Quasistatic " + direction)
+        .finallyDo(this::stop);
+  }
+
+  public Command sysIdShooterMotorDynamCommand(Direction direction) {
+    return shooterMotorSysIdRoutine.dynamic(direction).withName("SysId Shooter Motors Quasistatic " + direction)
+        .finallyDo(this::stop);
+  }
+
+  public void activeStop() {
+    if (!isActiveStopped) {
+      shooterLeftMotor.setControl(shooterMotorVelocity.withVelocity(0));
+      shooterRightMotor.setControl(shooterMotorVelocity.withVelocity(0));
+    }
+    isActiveStopped = true;
+  }
+
+  /**
+   * Gets the wrist position Zero is horizontal, up is positive
+   * 
+   * @return position in radians
+   */
+  public double getWristPosition() {
+    return Units.rotationsToRadians(actuatorEncoder.getPosition() + ENCODER_OFFSET);
+  }
+
+  public void actuatorUp(Measure<Voltage> volts) {
+    indexerMotorConfig(actuatorMotor, true);
+    actuatorMotor.setVoltage(volts.in(Volts));
+  }
+
+  public void actuatorDown(Measure<Voltage> volts) {
+    indexerMotorConfig(actuatorMotor, false);
+    actuatorMotor.setVoltage(volts.in(Volts));
+  }
+
+  /**
+   * Gets the wrist velocity in radians per second
+   * 
+   * @return wrist velocity in radians per second
+   */
+  public double getWristVelocity() {
+    return Units.rotationsToDegrees(actuatorEncoder.getVelocity());
+  }
+
+  public boolean checkWristPosition(double radiansToRotate) {

```

Comment:
I think it would be more useful to have a method that returns `true` if the wrist is at the previously requested velocity. Having it this way will be hard to deal with in a command. The same applies to the `checkShooterSpeed` method.